http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala

```diff
@@ -237,17 +241,29 @@ class ProxyRoutes(proxyService: ProxyService, corsSupport: CorsSupport, refererC
     for {
       ctx <- ev.ask[AppContext]
       apiCall = proxyService.proxyRequest(userInfo, googleProject, runtimeName, request)
-      _ <- metrics.incrementCounter("proxyRuntime")
       resp <- ctx.span.fold(apiCall)(span => spanResource[IO](span, "proxyRuntime").use(_ => apiCall))
+
+      headerMap: Map[String, String] = request.headers.map(header => (header.name(), header.value())).toMap
+      contentLength = Try(headerMap.get("content-length").map(_.toDouble)).toOption.flatten
+
+      _ <- if (request.uri.toString.endsWith(".ipynb") && request.method.equals(Method.PUT)) {
+        contentLength.traverse(size => metrics.gauge("proxy/notebooksSize", size))
```

Comment:
`traverse(...)` will be executed only if `contentLength` is a `Some(...)`. For instance, if `content-length` parsed from a request is not anything cast-able to `Double` then nothing will be recorded as a `notebooksSize` metric…which sounds reasonable to me but I wanted to make sure that was the intention.

Reply:
yeah i think that's okay, good catch though!